common/include/vowpalwabbit/common/rand48.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+// A quick implementation similar to drand48 for cross-platform compatibility
+#include <cmath>
+#include <cstdint>
+
+#include "vowpalwabbit/common/future_compat.h"
+
+//
+// NB: the 'ULL' suffix is not part of the constant it is there to
+// prevent truncation of constant to (32-bit long) when compiling
+// in a 32-bit env: warning: integer constant is too large for "long" type
+//
+#ifdef __clang__
+#  pragma clang diagnostic ignored "-Wc++11-long-long"
+#endif
+constexpr uint64_t CONSTANT_A = 0xeece66d5deece66dULL;
+constexpr uint64_t CONSTANT_C = 2147483647;
+
+constexpr int BIAS = 127 << 23;
+
+namespace VW
+{
+namespace common
+{
+inline float merand48(uint64_t& initial)
+{
+  static_assert(sizeof(int) == sizeof(float), "Floats and ints are converted between, they must be the same size.");
```

Comment:
I don't see `int` playing any part in this. We're taking the lower 32 bits of a `uint64_t`, shift/masking it and casting to a float. Really, this should be checking `sizeof(uint32_t) == sizeof(float)`

Reply:
Yes, that's true this really should be the actual type we are manipulating `int32_t`

Reply:
I'll fix this in master since this is a draft